File: 05_Sliding_Window/buy_and_sell_brute.py
"""
Link: https://leetcode.com/problems/best-time-to-buy-and-sell-stock
Brute Force Approach
Let prices[i] = [7,1,5,3,6,4]
"""

# brute force
def maxProfit(prices):

    # stores the maximum profit
    max_profit = 0

    for i in range(0, len(prices) - 1, 1):
        for j in range(i+1, len(prices), 1):
            profit = prices[j] - prices[i]
            if profit > max_profit:
                max_profit = profit
    
    return max_profit

    
print(maxProfit([7,1,5,3,6,4]))
print(maxProfit([7,6,4,3,1]))

File: 05_Sliding_Window/buyAndSellBrute.cpp
#include <bits/stdc++.h>
using namespace std;

int maxProfit(vector<int> prices) {
    int max_profit = 0;
    for(int i = 0; i < prices.size()-1; i++) {
        for(int j = i+1; j < prices.size(); j++) {
            int profit = prices[j] - prices[i];
            if(profit > max_profit) {
                max_profit = profit;
            }
        }
    }
    return max_profit;
}

int main() {
    cout << maxProfit({7,1,5,3,6,4}) << "\n";
}